### PR TITLE
Improve Blade-X Components

### PR DESCRIPTION
**Pull request by Shidersz**
| Question                | Answer
| ----------------------- | -----------------------
| Issue or Enhancement    | Enhancement
| License                 | MIT

#### What's in this PR?

Another set of improvements related to `blade-x` components (See #732 )

- Improve the **input-file** component to display better when using the size property  (now use pure `CSS`).
- Improve the **select2** component, now extends the **input-group-component**, supports a new `array` attribute for make the underlying plugin configuration and implements `CSS` workarounds when using the size attribute.
- Improve the **select-icon** component, renamed to **select-bs** and now extends the **input-group-component**. Also supports a new `array` attribute for make the underlying plugin configuration and implements workarounds when using the size attribute.
- Drop **option** component, there is no sense to create a component for something basic as an `option`. There was no benefit from it and restricted some features for option customization you can use from `bootstrap-select` plugin.

#### Checklist

- [x] I tested these changes.
- [x] I have linked the related issues.

### CODE REVIEW

File: resources/views/components/select2.blade.php [hunk 1]
```diff
@@ -1,22 +1,63 @@
...
     </script>
-@endsection
\ No newline at end of file
+@endpush
+
+{{-- Setup the height and font size of the plugin when using sm/lg sizes --}}
+{{-- NOTE: this may change with newer plugin versions --}}
+
+@once
+@push('css')
```
Comment (Shidersz) [inline resources/views/components/select2.blade.php:27]: @REJack Do you know if `AdminLte` package is implementing this type of support on the incoming versions. Maybe these workarounds to support plugins inside and input group and work nicely with `input-group-[sm/lg]` can be added to the package in a future...

Comment (REJack) [reply resources/views/components/select2.blade.php:27]: There is nothing planed for select2 inside input-group but I'll look over it and may bring iz with v3.1. At the moment our custom select2 style has a support for form-control-sm.

Comment (Shidersz) [reply resources/views/components/select2.blade.php:27]: Thanks @REJack , maybe for the particular case of `select2` those customization may be inside [select2-bootstrap4-theme](https://github.com/ttskch/select2-bootstrap4-theme) plugin. However, I had to implement a similar fix for the `input-file` component, and probably I will have to do the same for other form components.

**Comment by Shidersz** [comment]
@lacodimizer This is ready to be merged...

**Pull request merged**